public/app/plugins/datasource/dashboard/DashboardQueryEditor.test.tsx

```diff
@@ -68,33 +68,39 @@ describe('DashboardQueryEditor', () => {
   });
 
   it('does not show a panel with the SHARED_DASHBOARD_QUERY datasource as an option in the dropdown', () => {
-    render(
-      <DashboardQueryEditor
-        queries={mockQueries}
-        panelData={mockPanelData}
-        onChange={mockOnChange}
-        onRunQueries={mockOnRunQueries}
-      />
-    );
-    const select = screen.getByText('Choose panel');
-    userEvent.click(select);
+    act(() => {
```

Comment:
> The tests pass, but react complains about update to the component not being wrapped in act - not sure where exactly this is happening though. :(

pretty sure this is the same thing josh also hit in a different PR. i _think_ the problem goes something like this...

we have a react component that does something async.
we render the react component <sup>synchronously</sup>.
we then make some assertions <sup>synchronously</sup>.
test ends, and the update then fires <sup>asynchronously</sup> -> error

so the best way to fix it is to await on the asynchronous thing somehow. just looking at the code, it looks like there's a loading spinner that then goes away when the async call completes maybe? if so, you could do something like `await waitForElementToBeRemoved(() => screen.queryBy*(<loadingSpinnerSelector>))` as per https://kentcdodds.com/blog/fix-the-not-wrapped-in-act-warning#how-to-fix-the-act-warning

another hackier way to fix it is to ensure you use an async method in your test _somewhere_. then it seems to wait on the asynchronous calls. e.g. make the whole test async, and change `const select = screen.getByText('Choose panel');` to `const select = await screen.findByText('Choose panel');`, but that's definitely a bit of a hack and way less clear.

Reply:
forgot to actually write the pain point i was trying to make (🤦 ): we shouldn't need to wrap anything in our tests in `act()`, `react-testing-library` should handle that